django_cyverse_auth/models.py

```diff
@@ -150,71 +151,70 @@ def create_access_token(token_key, token_expire, issuer):
     return access_token
 
 
-def create_user_and_token(user_profile, token_key, token_expire=None, remote_ip=None, issuer=None):
+def get_or_create_user_and_token(user_profile, token_key, token_expire=None, remote_ip=None, issuer=None):
     """
     Given a user_profile (minimally) containing the keys: ['username', 'firstName', 'lastName', 'email'] and (optionally) a token UUID
     Create the user
     Create token for user
     return token
     """
-    get_or_create_user(user_profile['username'], user_profile)
-    auth_token = create_token(user_profile['username'], token_key, token_expire, remote_ip, issuer)
+    try:
+        get_or_create_user(user_profile['username'], user_profile)
+    except IntegrityError:
+        get_or_create_user(user_profile['username'], user_profile)
+    try:
+        auth_token = get_or_create_token(user_profile['username'], token_key, token_expire, remote_ip, issuer)
+    except IntegrityError:
+        auth_token = get_or_create_token(user_profile['username'], token_key, token_expire, remote_ip, issuer)
+
```

Comment:
The method body can be replaced by:
get_or_create_token(user_profile['username'], token_key, token_expire, remote_ip, issuer)

The better fix is to nuke this method and replace all occurences of `get_or_create_user_and_token` with `get_or_create_token`